src/libs/ReportActionsUtils.ts

```diff
@@ -966,28 +966,31 @@ function getOneTransactionThreadReportID(reportID: string, reportActions: OnyxEn
         return;
     }
 
-    const reportActionsArray = Object.values(reportActions ?? {});
+    const reportActionsArray = Array.isArray(reportActions) ? reportActions : Object.values(reportActions ?? {});
     if (!reportActionsArray.length) {
         return;
     }
 
     // Get all IOU report actions for the report.
-    const iouRequestTypes: Array<ValueOf<typeof CONST.IOU.REPORT_ACTION_TYPE>> = [
+    const iouRequestTypes = new Set<ValueOf<typeof CONST.IOU.REPORT_ACTION_TYPE>>([
```

Comment:
Will declaring this set outside of the function's body help here?

Reply:
I found the improvement to be very slight, but may be more noticeable on mobile devices